nautobot/utilities/views.py

```diff
@@ -151,3 +187,1051 @@ def get_return_url(self, request, obj=None):
+
+        # Add custom field headers, if any
+        if hasattr(self.queryset.model, "_custom_field_data"):
+            for custom_field in CustomField.objects.get_for_model(self.queryset.model):
+                headers.append(custom_field.name)
+                custom_fields.append(custom_field.name)
+
+        csv_data.append(",".join(headers))
+
+        # Iterate through the queryset appending each object
+        for obj in self.queryset:
+            data = obj.to_csv()
+
+            for custom_field in custom_fields:
+                data += (obj.cf.get(custom_field, ""),)
+
+            csv_data.append(csv_format(data))
+
+        return "\n".join(csv_data)
+
+    def validate_action_buttons(self, request):
+        """Verify actions in self.action_buttons are valid view actions."""
+
+        always_valid_actions = ("export",)
+        valid_actions = []
+        invalid_actions = []
+
+        for action in self.action_buttons:
+            if action in always_valid_actions or validated_viewname(self.queryset.model, action) is not None:
+                valid_actions.append(action)
+            else:
+                invalid_actions.append(action)
+        if invalid_actions:
+            messages.error(request, f"Missing views for action(s) {', '.join(invalid_actions)}")
+        return valid_actions
+
+    def handle_object_list_get(self, request):
+        model = self.queryset.model
+        content_type = ContentType.objects.get_for_model(model)
+
+        self.queryset = self.filterset(request.GET, self.queryset).qs
+
+        # Check for export template rendering
+        if request.GET.get("export"):
+            et = get_object_or_404(
+                ExportTemplate,
+                content_type=content_type,
+                name=request.GET.get("export"),
+            )
+            try:
+                return et.render_to_response(self.queryset)
+            except Exception as e:
+                messages.error(
+                    request,
+                    f"There was an error rendering the selected export template ({et.name}): {e}",
+                )
+
+        # Check for YAML export support
+        elif "export" in request.GET and hasattr(model, "to_yaml"):
+            response = HttpResponse(self.queryset_to_yaml(), content_type="text/yaml")
+            filename = f"nautobot_{self.queryset.model._meta.verbose_name_plural}.yaml"
+            response["Content-Disposition"] = f'attachment; filename="{filename}"'
+            return response
+
+        # Fall back to built-in CSV formatting if export requested but no template specified
+        elif "export" in request.GET and hasattr(model, "to_csv"):
+            response = HttpResponse(self.queryset_to_csv(), content_type="text/csv")
+            filename = f"nautobot_{self.queryset.model._meta.verbose_name_plural}.csv"
+            response["Content-Disposition"] = f'attachment; filename="{filename}"'
+            return response
```

Comment:
I hate this old pattern and I believe we can get rid of it entirely. This should be a job for a `Renderer` subclass and not hard-coded into a magic query string parameter.

For example: For [YAML](https://www.django-rest-framework.org/api-guide/renderers/#yaml), and [CSV](https://www.django-rest-framework.org/api-guide/renderers/#csv).